sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/BooleanSimplificationSuite.scala

```diff
@@ -330,4 +330,118 @@ class BooleanSimplificationSuite extends PlanTest with ExpressionEvalHelper with
       checkEvaluation(originalExpr, optimizedVal, inputRow)
     }
   }
+
+  test("SPARK-36665: Add more Not operator simplifications") {
+    // Using IsNull(e(inputs)) == IsNull(input1) or IsNull(input2) ... rules
+    checkCondition(IsNull(Not('e)), IsNull('e))
+    checkCondition(IsNotNull(Not('e)), IsNotNull('e))
+
```

Comment:
This test doesn't show the `Or` generated. Could you add another test there are more than one child expression?

Reply:
Ah, the comment is obsolete. Based on earlier reviews, we are not restricting that rules for UnaryExpression that does not create `Or`. I will update it. Thanks

Reply:
updated